src/library-authoring/add-content/AddContentContainer.test.tsx

```diff
@@ -89,8 +89,36 @@ describe('<AddContentContainer />', () => {
     const pasteButton = await screen.findByRole('button', { name: /paste from clipboard/i });
     fireEvent.click(pasteButton);
 
-    await waitFor(() => expect(axiosMock.history.post[0].url).toEqual(pasteUrl));
+    await waitFor(() => {
+      expect(axiosMock.history.post[0].url).toEqual(pasteUrl);
+      expect(mockShowToast).toHaveBeenCalledWith('There was an error pasting the content.');
+    });
+  });
+
+  it('should handle failure to paste content and show server error if available', async () => {
+    const { axiosMock, mockShowToast } = initializeMocks();
+    // Simulate having an HTML block in the clipboard:
+    mockClipboardHtml.applyMock();
+
+    const errMsg = 'Libraries do not support this type of content yet.';
+    const pasteUrl = getLibraryPasteClipboardUrl(libraryId);
+
+    // eslint-disable-next-line prefer-promise-reject-errors
+    axiosMock.onPost(pasteUrl).reply(() => Promise.reject({
+      customAttributes: {
+        httpErrorStatus: 400,
+        httpErrorResponseData: JSON.stringify({ block_type: errMsg }),
+      },
+    }));
```

Comment:
Nit: Did you try the `createAxiosError` function from the `testUtils.tsx`?

Reply:
@rpenido Doesn't seem to match the error response from server when paste fails. We don't really need it now but it is nice to have for other paste related errors.